### PR TITLE
Allow accessing internal raw byte buffer

### PR DESCRIPTION
**Pull request by zonyitoo**
* Accessing internal raw byte buffer
* Constructing `d128` with raw byte buffer
* Endianness conversion

### CODE REVIEW

File: src/dec128.rs [hunk 1]
```diff
@@ -108,8 +108,10 @@ impl serde::de::Visitor for d128Visitor {
         where E: serde::de::Error
     {
         let d = match d128::from_str(s) {
-            Ok(d) => { d }
-            Err(_) => { return Err(E::custom("Failed to convert to d128")); }
+            Ok(d) => d,
+            Err(_) => {
+                return Err(E::custom("Failed to convert to d128"));
+            }
```
Comment (alkis) [inline src/dec128.rs]: Let's remove the formatting changes since they are unrelated to this PR.

File: src/dec128.rs [hunk 2]
```diff
@@ -456,6 +458,16 @@ impl d128 {
         CTX.with(|ctx| f(&mut ctx.borrow_mut()))
     }
 
+    /// Creates from raw bytes
```
Comment (alkis) [inline src/dec128.rs]: I think a better interface is to say that from_bytes() is machine dependent. The user must handle endianess manually if it cares to do so. As such the documentation should be:

```rust
/// Creates a d128 from raw bytes. Endianess is host dependent.
pub unsafe fn from_raw_bytes(bytes: [u8; 16]) -> d128 ...
```

File: src/dec128.rs [hunk 3]
```diff
@@ -456,6 +458,16 @@ impl d128 {
...
+    pub unsafe fn from_bytes(bytes: [u8; 16]) -> d128 {
+        d128 { bytes: bytes }
+    }
+
+    /// Gets raw bytes
```
Comment (alkis) [inline src/dec128.rs]: Similarly:

```rust
/// Returns raw bytes for this d128. Endianess is host dependent.
pub fn as_bytes(&self) -> &[u8] ...
```

File: src/dec128.rs [hunk 4]
```diff
@@ -786,6 +798,48 @@ impl d128 {
     pub fn is_zero(&self) -> bool {
         unsafe { decQuadIsZero(self) != 0 }
     }
+
```
Comment (alkis) [inline src/dec128.rs]: Remove all of these. The user should handle this on its own.

Comment (zonyitoo) [inline src/dec128.rs]: Hmm, why? As we can see in libstd, `i16`, `u16`, `i32`, ... all provides `from_*` and `to_*` functions for endianness conversion.

File: src/dec128.rs (hunk 3 above)
Comment (zonyitoo) [inline src/dec128.rs]: Then it should be named as `as_raw_bytes`. Accepted.

File: src/dec128.rs (hunk 2 above)
Comment (zonyitoo) [inline src/dec128.rs]: Accepted.

File: src/dec128.rs (hunk 4 above)
Comment (alkis) [inline src/dec128.rs]: There is no from_le and to_le for floating point types. This is a floating point type. le/be for it is not well defined.

Comment (zonyitoo) [inline src/dec128.rs]: Hmm, alright. I will remove them.

**Comment by zonyitoo** [comment]
@alkis Please check the latest commit.

**Comment by zonyitoo** [comment]
Thanks a lot!

**Comment by alkis** [comment]
Thank you for the contribution!

**Comment by zonyitoo** [comment]
Maybe you have forgotten to click the "Merge" button. @alkis
